internal/cow/cow.go

```diff
@@ -88,6 +88,10 @@ type Container interface {
 	// container to be terminated by some error condition (including calling
 	// Close).
 	Wait() error
+	// WaitChannel returns the wait channel of the container
+	WaitChannel() <-chan struct{}
+	// WaitError returns the container termination error
+	WaitError() error
```

Comment:
Would be good to add a comment here that `WaitError` should _only_  be called _after_ the channel from `WaitChannel` is closed. Otherwise it is not thread safe.